### PR TITLE
collect-info: collect longhorn support bundle

### PR DESCRIPTION
**Pull request by andrewd-zededa**
Run in kube container for k3s access.

### CODE REVIEW

**Comment by andrewd-zededa** [comment]
The arm builds failed like this: 

> #22 [build  7/53] RUN su builder -c 'cd /musl && abuild checksum && abuild -r'
> #22 0.416 >>> musl: Updating the sha512sums in APKBUILD...
> #22 0.829 >>> musl: Building /musl 1.2.3-r0 (using abuild 3.9.0-r0) started Fri, 19 Jul 2024 23:39:41 +0000
> #22 0.890 >>> musl: Checking sanity of /musl/APKBUILD...
> #22 1.134 >>> musl: Analyzing dependencies...
> #22 1.282 >>> musl: Installing for build: build-base
> Error: error building "lfedge/eve-debug:9a72bf847fa0f0c2f4821f14176752d730f0b0dd": error building for arch arm64: failed to solve: process "/dev/.buildkit_qemu_emulator /bin/sh -c su builder -c 'cd /musl && abuild checksum && abuild -r'" did not complete successfully: exit code: 1
> 2024/07/19 23:39:42 error during command execution: error building "lfedge/eve-debug:9a72bf847fa0f0c2f4821f14176752d730f0b0dd": error building for arch arm64: failed to solve: process "/dev/.buildkit_qemu_emulator /bin/sh -c su builder -c 'cd /musl && abuild checksum && abuild -r'" did not complete successfully: exit code: 1
> make: *** [Makefile:1030: eve-debug] Error 1
> Error: Process completed with exit code 2.

The first failure before this was an apk add again.  This was seen on some CI runs lately but this time only ca-certificates vs a longer list of packages on other PR runs (https://github.com/lf-edge/eve/pull/4074#issuecomment-2231359759, https://github.com/lf-edge/eve/actions/runs/9913215972/job/27390017159?pr=4074#step:9:860)

#16 4.516 Executing busybox-1.35.0-r17.trigger
#16 4.522 ERROR: busybox-1.35.0-r17.trigger: script exited with error 127
#16 4.523 Executing ca-certificates-20240226-r0.trigger
#16 4.526 ERROR: ca-certificates-20240226-r0.trigger: script exited with error 127

**Comment by andrewd-zededa** [comment]
Rebase on master

**Review by milan-zededa — APPROVED** [review]
Still draft or ready for merge?

File: pkg/debug/scripts/collect-info.sh [hunk 1]
```diff
@@ -369,6 +369,23 @@ collect_kube_info()
...
+    type=$(cat /run/eve-hv-type)
+    if [ "$type" != "kubevirt" ]; then
+        return
+    fi
+    echo "- Collecting Longhorn specific info"
+    {
+        echo "  - longhorn support bundle "
+        # Longhorn issue reports have seen hangs generating this data
+        # when nodes are down, or due to longhorn bugs.
+        # Give up after 5min, and allow remaining system data to be collected.
```
Comment (milan-zededa) [inline pkg/debug/scripts/collect-info.sh:384]: How large can this be?
5 min. seems like a lot of time compared to a typical duration of collect-info.sh. If stuck, user would likely loose patience and kill the whole script. Can we lower the timeout to seconds?

Comment (andrewd-zededa) [reply pkg/debug/scripts/collect-info.sh:384]: Some of the variability can be from the pull of the support bundle image.  On a three node system I just saw 132 seconds for an initial run due to also pulling the support bundle image, and a later run was only 45 seconds.   This is a good point i'll cleanup the comments a bit and change the echo to inform the user to be patient as this step can take a longer time.

Comment (andrewd-zededa) [reply pkg/debug/scripts/collect-info.sh:386]: @milan-zededa changes added to help address users wait time expectation.  I'm afraid the timeout would have to be dropped to something closer to 30-60 seconds to fit in with the existing time spans.  To get closer to this (with room to spare) we will need to start by including the support image in the system, I will investigate this option for a later PR.

**Comment by andrewd-zededa** [comment]
@milan-zededa this is ready now, I had left it in draft due to the unrelated CI/build issues.

**Pull request merged**